app/models/metric/helper.rb

```diff
@@ -64,6 +72,19 @@ def self.next_daily_timestamp(ts, tz = nil)
     nearest_daily_timestamp(ts + 1.day, tz)
   end
 
+  # @option range :start_date start of the range (if not present, sets to end_date - days)
+  # @option range :end_date end of time range (default: now)
+  # @option range :days number of days for range (default: 20)
+  # @return Range<DateTime,DateTime>
+  def self.timestamp_from_hash(range)
+    return range unless range.kind_of?(Hash)
+    range[:days] ||= 20
```

Comment:
This should probably just use a local: there's no reason for us to be mutating the hash we were passed. Even if we're planning to kill off this hash (are we?), in the meantime it'll be slightly easier to reason about without hidden side-effects.